### PR TITLE
replace the deprecated `provision-with-micromamba` with `setup-micromamba`

### PR DESCRIPTION
**Pull request by keewis**

### CODE REVIEW

**Comment by keewis** [comment]
the `mypy` change (ignoring `unused-ignore`) could use a review – I barely know what I'm doing when it comes to typing – but otherwise this should be ready for merging

**Comment by dcherian** [comment]
I think mypy is telling you to delete that ignore comment?

**Comment by keewis** [comment]
As far as I can tell, `numpy` improved its typing support, but because we still support older versions the type checking would fail with those (hence the `unused-ignore` ignore). Not sure if this applies here, as well.

**Comment by dcherian** [comment]
I would just delete it. Maintaining backwards compatibility for typing is a bit too much :)

File: flox/core.py [hunk 1]
```diff
@@ -1622,7 +1622,7 @@ def _convert_expected_groups_to_index(
                 out.append(ex)
         elif ex is not None:
             if isbin_:
-                out.append(pd.IntervalIndex.from_breaks(ex))  # type: ignore [arg-type] # TODO: what do we want here?
+                out.append(pd.IntervalIndex.from_breaks(ex))  # type: ignore [arg-type,unused-ignore] # TODO: what do we want here?
```
Comment (dcherian) [inline flox/core.py:1625]:
```suggestion
                out.append(pd.IntervalIndex.from_breaks(ex))
```

**Pull request merged**